include/graaflib/algorithm/cycle_detection.h

```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#include <graaflib/graph.h>
+
+namespace graaf::algorithm {
+
+/*
+ * @breif Traverses the graph and checks for cycles.
+ *
+ * @param graph The graph to traverse.
+ */
```

Comment:
[very minor]: last typo
```suggestion
/*
 * @brief Traverses the graph and checks for cycles.
 *
 * @param graph The graph to traverse.
 */
```